dynamiq/memory/backend/pinecone.py

```diff
@@ -1,153 +1,150 @@
 import uuid
-from typing import Any
-
-from pinecone import ServerlessSpec
 
 from dynamiq.components.embedders.base import BaseEmbedder
 from dynamiq.connections import Pinecone as PineconeConnection
 from dynamiq.memory.backend.base import MemoryBackend
 from dynamiq.prompts import Message
+from dynamiq.storages.vector.pinecone import PineconeVectorStore
+from dynamiq.storages.vector.pinecone.pinecone import PineconeIndexType
+from dynamiq.types import Document
 
 
 class PineconeError(Exception):
     """Base exception class for Pinecone-related errors."""
-
     pass
 
 
 class Pinecone(MemoryBackend):
-    """Pinecone implementation of the memory storage backend."""
-
     name = "Pinecone"
 
     def __init__(
         self,
         connection: PineconeConnection,
         embedder: BaseEmbedder,
-        cloud: str,
-        region: str,
         index_name: str = "conversations",
+        namespace: str = "default",
+        index_type: PineconeIndexType | None = None,
```

Comment:
with `create_if_not_exist=True`, `index_type` can't be Null. Still we handle this case and user would see error during init if index not exists and `index_type` is Null. So probably could ignore this

Reply:
Alternatively, set the default value of index_type to 'serverless'.
WDYT?